core/signer/signer.go

```diff
@@ -63,10 +63,26 @@ func New(realm string,
 	}
 }
 
-// StartSigning advance epochs once per interval.
-func (s *Signer) StartSigning(ctx context.Context, interval time.Duration) {
-	for range time.NewTicker(interval).C {
-		if err := s.CreateEpoch(ctx); err != nil {
+// StartSigning advance epochs once per minInterval, if there were mutations, and at least once per
+// maxElapsed minIntervals.
+func (s *Signer) StartSigning(ctx context.Context, minInterval time.Duration, maxInterval time.Duration) {
+	tc := time.Tick(minInterval)
+	processEpochs(ctx, tc, maxInterval, s.CreateEpoch)
+}
+
+func processEpochs(ctx context.Context, tc <-chan time.Time, maxElapsed time.Duration, sign func(ctx context.Context, enforce bool) error) {
```

Comment:
The only reason I only created `processEpochs` to make it possible to test the underlying logic without the need of creating a whole `Signer` (which depends on a lot of other running objects, like the DB, a mutator, transaction factory...). Other than making testing convinient we don't need, I guess.